editor/sidebar/post-visibility/index.js

```diff
@@ -111,12 +129,17 @@ class PostVisibility extends Component {
 					aria-expanded={ this.state.opened }
 					className="editor-post-visibility__toggle button-link"
 					onClick={ this.toggleDialog }
+					ref={ toggle => this.toggle = toggle }
 				>
 					{ getVisibilityLabel( visibility ) }
 				</button>
 
 				{ this.state.opened &&
-					<div className="editor-post-visibility__dialog">
+					<div className="editor-post-visibility__dialog"
+						onKeyDown={ this.handleKeyDown }
```

Comment:
Hm I think the rule makes perfectly sense for example to avoid to use `<div>` or `<span>` elements as UI controls instead of buttons. Re: the warning: it should probably be evaluated on a case by case basis. Our `onKeyDown` here is actually used for the radio buttons and input field. Maybe the rule should be improved to check if a `<div>` is focusable and actually able to receive a keyboard event. Anyways, it's good to have it.